device/src/drivers/ble/mesh/model/sensor/mod.rs

```diff
@@ -82,7 +82,9 @@ pub trait SensorConfig: Clone {
 }
 
 pub trait SensorData {
-    fn decode(&mut self, property: PropertyId, data: &[u8]) -> Result<(), ParseError>;
+    fn decode(property: PropertyId, data: &[u8]) -> Result<Self, ParseError>
```

Comment:
Looking at how this is being used now, SensorData can't have more than 1 property, which won't work in the general case. I think we need to modify the trait back and add a default constraint:
```
pub trait SensorData: Default {
    fn decode(&mut self, property: PropertyId, data: &[u8]) -> Result<(), ParseError>
}
```

Reply:
Yeah, I wasn't happy with that solution. I reworked it to use Default trait and now it looks much nicer.